### ISSUE TITLE
Button push only being registered between loops

### ISSUE THREAD

**Issue by Uberibsen** [Uberibsen/AstralisLEDSign#1]
In the current code, the sequence button only increments `sequenceNumber` by one in between the loops when pressed. 

### Possible solution
The ideal solution would be that no matter where the effect is in its duration (for loops, i.e.), the button push would be registered.  The fix would have to implement the code below into the for loops running in each effect.

```  
buttonState = digitalRead(buttonPin);

  if (buttonState != lastButtonState) {
    if (buttonState == HIGH) {
      sequenceNumber++;
    }
    delay(50);
  }
  // save the current state as the last state, for next time through the loop
  lastButtonState = buttonState;
``` 

This iteration of the code runs that the individual effect would have to finish before it goes back around to register a button push.

**Comment by Duckapple** [comment]
The problem with the unresponsive button is the fact that every light routine is carried out in its entirety before returning to the core loop again. 
To alleviate this, you should try to query and update some state variable outside of the loop instead of iterating in the for loops.

An example:
```c++
//// before ////
void routine() {
    for (int i = 0; i < 10000; i++) { // Runs for 10 * 10000 = 100 seconds
        blink_light(i % 2); // alternates 0 and 1
        delay(10);
    }
}

void loop() {
    checkButton();
    routine();
}

//// after ////
int i = 0;

void routine() {
    if (i >= 10000) { // check whether to reset 'i'
        i = 0;
    } // could also just do the line: i = i % 10000
    blink_light(i % 2);
    i++;
}

void loop() {
    checkButton();
    routine();
    delay(10); // moved to loop for a global clock
}

**Comment by Uberibsen** [comment]
Closed by ce3b18f2a623b125571f4222db1db631e753e06b
